cpu.go

```diff
@@ -1,18 +1,38 @@
 package sqrl
 
 import (
+	"encoding/json"
 	"fmt"
 	"log"
+	"runtime"
+	"strings"
 
 	"github.com/shirou/gopsutil/cpu"
 )
 
-func CPUInfo() error {
+type CPUInfo struct {
+	CPU   int    `json:"cpu"`
+	Cores int32  `json:"cores"`
+	Speed string `json:"speed"`
+	Model string `json:"model"`
+}
+
+func GetCPUInfo() {
 	info, err := cpu.Info()
 	if err != nil {
-		log.Printf("error getting cpu: %v", err)
-		return err
+		log.Fatal(err)
+	}
+	mInfo := strings.Split(info[0].ModelName, "@") // mInfo contains speed and model
+	c := CPUInfo{
+		CPU:   runtime.NumCPU(),
+		Cores: info[0].Cores,
+		Speed: mInfo[1],
+		Model: mInfo[0],
+	}
+
+	j, err := json.Marshal(c)
+	if err != nil {
+		log.Fatal(err)
 	}
-	fmt.Println(info)
-	return nil
+	fmt.Println(string(j))
```

Comment:
instead of just printing this out, would you mind returning this?  Change the fucntion to return a CPUInfo strruct. 

Also, please don't log.Fatal(err).  Pass that error up to the caller so that it can be handled there.